crates/miden-objects/src/block/proven_block.rs

```diff
@@ -50,6 +50,9 @@ pub struct ProvenBlock {
     /// The aggregated and flattened transaction headers of all batches in the order in which they
     /// appeared in the proposed block.
     transactions: OrderedTransactionHeaders,
```

Comment:
I would introduce the `BlockBody` struct in this PR which would consist of these 4 fields.